File: app/src/main/java/mb00/android/codehub/ui/user/inject/UserPulseFragmentModule.java
package mb00.android.codehub.ui.user.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.user.interactor.UserInteractor;
import mb00.android.codehub.ui.user.router.UserRouter;
import mb00.android.codehub.ui.user.view.UserPulseFragment;
import mb00.android.codehub.ui.user.viewmodel.UserPulseViewModel;


@Module
public class UserPulseFragmentModule {

    @Provides
    UserPulseFragment provideUserPulseFragment(UserPulseFragment fragment) {
        return fragment;
    }

    @Provides
    UserPulseViewModel provideUserPulseViewModel(UserInteractor interactor, UserRouter router) {
        return new UserPulseViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/inject/UserPulseFragmentModule.kt
package mb00.android.codehub.ui.user.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.user.interactor.UserInteractor
import mb00.android.codehub.ui.user.router.UserRouter
import mb00.android.codehub.ui.user.view.UserPulseFragment
import mb00.android.codehub.ui.user.viewmodel.UserPulseViewModel


@Module
class UserPulseFragmentModule {

    @Provides
    internal fun provideUserPulseFragment(fragment: UserPulseFragment): UserPulseFragment {
        return fragment
    }

    @Provides
    internal fun provideUserPulseViewModel(interactor: UserInteractor, router: UserRouter): UserPulseViewModel {
        return UserPulseViewModel(interactor, router)
    }

}
